### ISSUE TITLE
Using type application with a function call gets a wrong error message

### ISSUE THREAD

**Issue by hyperkai** [python/mypy#20927]
*Notes:
- mypy test.py
- mypy 1.19.1
- Python 3.14.0

Using type application with a function call gets the error message as shown below:

```python
def func[T](x: T) -> T:
    return x

func[int](100) # Error
```

> error: Type application is only supported for generic classes 

But type application is also supported for generic type aliases as shown below:

```python
type TA[T] = list[T]

v: TA[int] # No error
```

So the error message should be as shown below:

> error: Type application is only supported for generic classes and type aliases
